src/omerocrate/uploader.py

```diff
@@ -21,6 +23,95 @@
 Namespaces = dict[str, URIRef]
 Variables = dict[str, Identifier]
 
+
+class SegmentationUploader(BaseModel, arbitrary_types_allowed=True):
+    """
+    Class that handles uploading of segmentation masks to OMERO.
+    Users can subclass this to customise segmentation upload behavior.
+    """
+    conn: gateway.BlitzGateway
+    "OMERO connection object, typically obtained using [`from_env`][omerocrate.gateway.from_env]"
+
+    def process_segmentation(self, segmentation_path: Path, image: gateway.ImageWrapper) -> None:
+        """
+        Load segmentation mask and upload to OMERO for the given image URI.
+        Users should override this method to implement segmentation upload logic.
+        """
+        raise NotImplementedError("process_segmentation() must be implemented in a subclass")
+
+
+class OmeNgffUploader(SegmentationUploader):
+    """
+    Subclass of SegmentationUploader that uses Glencoe's ROI_Converter_NGFF to upload segmentations.
+    Assumes that the segmentation file is a CSV with a 'polygon' or 'geometry' column containing WKT
+    polygons, and an 'object' or 'id' column for object identifiers.
+    Requires ROI_Converter_NGFF to be installed and accessible in the Python environment.
+    """
+
+    def __init__(self, conn: gateway.BlitzGateway):
+        if importlib.util.find_spec("ROI_Converter_NGFF") is None:
+            raise ImportError("ROI_Converter_NGFF is required for OmeNgffUploader.")
+        super().__init__(conn=conn)
```

Comment:
I didn't think of this earlier, but redefining constructors isn't great in Pydantic. How about a model validator `mode="after"`?